assets/js/plugin-check-admin.js

```diff
@@ -207,4 +213,85 @@
 
 		return data;
 	}
+
+	/**
+	 * Renders results for each check on the page.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} results The results object.
+	 */
+	function renderResults( results ) {
+		const { errors, warnings } = results;
+
+		// Render errors and warnings for files.
+		for ( const file in errors ) {
+			if ( warnings[ file ] ) {
+				renderFileResults( file, errors[ file ], warnings[ file ] );
+				delete warnings[ file ];
+			} else {
+				renderFileResults( file, errors[ file ], [] );
+			}
+		}
+
+		// Render remaining files with only warnings.
+		for ( const file in warnings ) {
+			renderFileResults( file, [], warnings[ file ] );
+		}
+	}
+
+	/**
+	 * Renders the file results table.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} file     The file name for the results.
+	 * @param {Object} errors   The file errors.
+	 * @param {Object} warnings The file warnings.
+	 */
+	function renderFileResults( file, errors, warnings ) {
+		const tableTemplate = wp.template( 'plugin-check-results-table' );
```

Comment:
Parsing the template here will result in the same template being parsed several times, we should avoid that.